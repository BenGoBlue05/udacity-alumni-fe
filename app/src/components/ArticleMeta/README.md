## ArticleMeta Component
A component that shows article tags and any other metadata

### Example

```js
<ArticleMeta article={article} />
```

### Props

| Prop          | Type     | Default     | Possible Values
| ------------- | -------- | ----------- | ---------------------------------------------
| **article**    | Object   |             | Article containing expected shape, as defined by schema.


### Other Information
NOTE: The tags are being faked right now because the schema does not return tags.  We need to add this.
