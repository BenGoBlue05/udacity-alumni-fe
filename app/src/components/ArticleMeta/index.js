import React, { PropTypes } from 'react';
import styles from './index.module.scss';
import cssModules from 'react-css-modules';
import Tags from 'grommet-udacity/components/Tags';
import Tag from 'grommet-udacity/components/Tag';
import Box from 'grommet-udacity/components/Box';
import Section from 'grommet-udacity/components/Section';
import Article from 'grommet-udacity/components/Article';
import Heading from 'grommet-udacity/components/Heading';
import Menu from 'grommet-udacity/components/Menu';
import SocialShare from 'grommet-udacity/components/SocialShare';
import Anchor from 'grommet-udacity/components/Anchor';

const ArticleMeta = ({
  article,
}) => (
  <Article className={styles.panel}>
    {article.tags && article.tags.length > 0 &&
      <Section>
        <Heading tag="h2" align="center">
          Article Tags
        </Heading>
        <Box
          justify="center"
          align="center"
          pad={{ vertical: 'medium' }}
        >
          <Tags align="center" justify="center">
            {article.tags.map((item, i) =>
              <Anchor
                key={i}
                href={`/archive?tag=${encodeURIComponent(item.tag)}`}
              >
                <Tag label={item.tag} />
              </Anchor>
            )}
          </Tags>
        </Box>
      </Section>
    }
    <Section>
      <Heading tag="h2" align="center">
        Share This Article
      </Heading>
      <Box align="center">
        <Menu
          inline
          direction="row"
          responsive={false}
        >
          <SocialShare
            a11yTitle="Go to Twitter to Share this article"
            type="twitter"
            link={`http://udacity-alumni.com/articles/${article.id}`}
            text={article.title}
          />
          <SocialShare
            a11yTitle="Go to Facebook to Share this article"
            type="facebook"
            link={`http://udacity-alumni.com/articles/${article.id}`}
            text={article.title}
          />
          <SocialShare
            a11yTitle="Go to linkedin to Share this article"
            type="linkedin"
            link={`http://udacity-alumni.com/articles/${article.id}`}
            text={article.title}
          />
          <SocialShare
            a11yTitle="Go to google to Share this article"
            type="google"
            link={`http://udacity-alumni.com/articles/${article.id}`}
            text={article.title}
          />
        </Menu>
      </Box>
    </Section>
  </Article>
);

ArticleMeta.propTypes = {
  article: PropTypes.shape({
    tags: PropTypes.array.isRequired,
    title: PropTypes.string.isRequired,
  }),
};

export default cssModules(ArticleMeta, styles);
