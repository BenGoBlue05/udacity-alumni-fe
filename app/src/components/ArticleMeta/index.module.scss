.panel {
  position: relative;
  width: 100%;
  max-width: 750px;
  background: #fff;
  box-shadow: 0 2px 4px 0 rgba(167, 157, 144, 0.52);
  padding: 60px !important;
}
