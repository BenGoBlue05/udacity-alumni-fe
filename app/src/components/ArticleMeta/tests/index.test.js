import ArticleMeta from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<ArticleMeta />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <ArticleMeta
        article={{
          title: 'Awesome post',
          tags: ['awesomeness'],
        }}
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
