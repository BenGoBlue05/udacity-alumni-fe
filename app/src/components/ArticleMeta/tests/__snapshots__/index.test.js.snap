exports[`<ArticleMeta /> should render with default props 1`] = `
<Article
  direction="column"
  pad="none">
  <Section
    pad={
      Object {
        "vertical": "medium"
      }
    }>
    <Heading
      align="center"
      tag="h2">
      Article Tags
    </Heading>
    <Box
      align="center"
      announce={false}
      direction="column"
      focusable={true}
      justify="center"
      pad={
        Object {
          "vertical": "medium"
        }
      }
      responsive={true}
      tag="div">
      <Tags
        align="center"
        direction="row"
        justify="center">
        <Anchor
          href="/archive?tag=undefined"
          tag="a">
          <Tag />
        </Anchor>
      </Tags>
    </Box>
  </Section>
  <Section
    pad={
      Object {
        "vertical": "medium"
      }
    }>
    <Heading
      align="center"
      tag="h2">
      Share This Article
    </Heading>
    <Box
      align="center"
      announce={false}
      direction="column"
      focusable={true}
      pad="none"
      responsive={true}
      tag="div">
      <Menu
        closeOnClick={true}
        direction="row"
        dropAlign={
          Object {
            "left": "left",
            "top": "top"
          }
        }
        inline={true}
        pad="none"
        responsive={false}>
        <SocialShare
          a11yTitle="Go to Twitter to Share this article"
          link="http://udacity-alumni.com/articles/undefined"
          text="Awesome post"
          title=""
          type="twitter" />
        <SocialShare
          a11yTitle="Go to Facebook to Share this article"
          link="http://udacity-alumni.com/articles/undefined"
          text="Awesome post"
          title=""
          type="facebook" />
        <SocialShare
          a11yTitle="Go to linkedin to Share this article"
          link="http://udacity-alumni.com/articles/undefined"
          text="Awesome post"
          title=""
          type="linkedin" />
        <SocialShare
          a11yTitle="Go to google to Share this article"
          link="http://udacity-alumni.com/articles/undefined"
          text="Awesome post"
          title=""
          type="google" />
      </Menu>
    </Box>
  </Section>
</Article>
`;
